main/src/ca/uwaterloo/flix/language/phase/Resolver.scala

```diff
@@ -1481,4 +1482,65 @@ object Resolver extends Phase[NamedAst.Root, ResolvedAst.Root] {
     Name.NName(sp1, idents, sp2)
   }
 
+  /**
+   * Creates the synthetic type classes `Eq` and `Ord`.
+   */
+  private def mkSynthClasses()(implicit flix: Flix): Map[Symbol.ClassSym, ResolvedAst.Class] = {
+    val eqTypeClass = mkEqClass()
+    val ordTypeClass = mkOrdClass(eqTypeClass)
+
+    Map(eqTypeClass.sym -> eqTypeClass, ordTypeClass.sym -> ordTypeClass)
+  }
+
+  /**
+   * Creates the synthetic type class `Eq`.
+   */
+  private def mkEqClass()(implicit flix: Flix): ResolvedAst.Class = {
+    val classSym = Symbol.mkClassSym(Name.RootNS, mkSynthIdent("Eq"))
+
+    val tparamType = Type.freshVar(Kind.Star)
+    val tparam = ResolvedAst.TypeParam(mkSynthIdent("a"), tparamType, SourceLocation.Generated)
+    val equalsSig = Sig(doc = synthDoc,
+      ann = Nil,
+      mod = Ast.Modifiers(List(Ast.Modifier.Synthetic)),
+      sym = Symbol.mkSigSym(classSym, mkSynthIdent("equals")),
+      tparams = List(tparam),
+      fparams = List(ResolvedAst.FormalParam(Symbol.freshVarSym(), Ast.Modifiers.Empty, tparamType, SourceLocation.Generated)),
+      sc = Scheme.generalize(Type.mkPureUncurriedArrow(List(tparamType, tparamType), Type.Bool)),
+      eff = Type.Pure,
+      loc = SourceLocation.Generated)
+
+    ResolvedAst.Class(classSym, List(tparam), List(equalsSig), Nil)
+  }
+
+  /**
+   * Creates the synthetic type class `Ord`, using the given `Eq` type class as a superclass.
+   */
+  private def mkOrdClass(eqTypeClass: ResolvedAst.Class)(implicit flix: Flix): ResolvedAst.Class = {
+    val classSym = Symbol.mkClassSym(Name.RootNS, mkSynthIdent("Ord"))
+
+    val tparamType = Type.freshVar(Kind.Star)
+    val tparam = ResolvedAst.TypeParam(mkSynthIdent("a"), tparamType, SourceLocation.Generated)
+    val leqSig = Sig(doc = synthDoc,
+      ann = Nil,
+      mod = Ast.Modifiers(List(Ast.Modifier.Synthetic)),
+      sym = Symbol.mkSigSym(classSym, mkSynthIdent("leq")),
+      tparams = List(tparam),
+      fparams = List(ResolvedAst.FormalParam(Symbol.freshVarSym(), Ast.Modifiers.Empty, tparamType, SourceLocation.Generated)),
+      sc = Scheme.generalize(Type.mkPureUncurriedArrow(List(tparamType, tparamType), Type.Bool)),
+      eff = Type.Pure,
+      loc = SourceLocation.Generated)
+
+    ResolvedAst.Class(classSym, List(tparam), List(leqSig), List(eqTypeClass))
+  }
+
+  /**
+   * Creates a synthetic Ident.
+   */
+  private def mkSynthIdent(name: String): Name.Ident = Name.Ident(SourcePosition.Unknown, name, SourcePosition.Unknown)
+
+  /**
+   * Synthetic documentation.
+   */
+  private val synthDoc: Ast.Doc = Ast.Doc(List(), SourceLocation.Generated) // MATT probably want real docs
```

Comment:
Ultimately I want the compiler to know about these, but they would be declared in actual Flix code (with their comments there)